src/common/Model/Config/AttributeConfig.php

```diff
@@ -2,93 +2,127 @@
 
-    private Json $json;
-    private DataProvider $dataProvider;
-    private CustomAttributeConfig $customAttributeConfig;
-
-    private bool $useVariantProducts;
+    /**
+     * @var array
+     */
     private array $productAttributes;
+    /**
+     * @var array
+     */
     private array $variantAttributes;
+    /**
+     * @var array
+     */
     private array $allAttributes;
+    /**
+     * @var array
+     */
     private array $booleanAttributes;
+    /**
+     * @var array
+     */
     private array $productAttributeCodes;
+    /**
+     * @var array
+     */
     private array $variantAttributeCodes;
-    private array $searchableAttributes;
+    /**
+     * @var array
+     */
     private array $staticAttributes;
+    /**
+     * @var array
+     */
     private array $attributesWithFredhopperType;
+    /**
+     * @var array
+     */
     private array $eavAttributesByType;
+    /**
+     * @var array
+     */
     private array $siteVariantAttributes;
+    /**
+     * @var array
+     */
+    private array $attributeNameMapping;
 
+    /**
+     * @param GeneralConfig $generalConfig
+     * @param ScopeConfigInterface $scopeConfig
+     * @param Json $json
+     * @param CustomAttributeConfig $customAttributeConfig
+     * @param AttributeDataProvider $attributeDataProvider
+     */
     public function __construct(
-        Context $context,
-        Resolver $localeResolver,
-        StoreManagerInterface $storeManager,
-        Json $json,
-        DataProvider $dataProvider,
-        CustomAttributeConfig $customAttributeConfig
+        private readonly GeneralConfig $generalConfig,
+        private readonly ScopeConfigInterface $scopeConfig,
+        private readonly Json $json,
+        private readonly CustomAttributeConfig $customAttributeConfig,
+        private readonly AttributeDataProvider $attributeDataProvider
     ) {
-        parent::__construct($context, $localeResolver, $storeManager);
-        $this->json = $json;
-        $this->dataProvider = $dataProvider;
-        $this->customAttributeConfig = $customAttributeConfig;
     }
 
     /**
+     * Gets whether to use variant products
+     *
      * @return bool
      */
     public function getUseVariantProducts(): bool
     {
-        if (!isset($this->useVariantProducts)) {
-            $this->useVariantProducts = $this->scopeConfig->isSetFlag(self::XML_PATH_USE_VARIANT_PRODUCTS);
-        }
-        return $this->useVariantProducts;
+        return $this->scopeConfig->isSetFlag(self::XML_PATH_USE_VARIANT_PRODUCTS);
     }
 
     /**
+     * Get all configured product attributes
+     *
      * @return array
      */
     public function getProductAttributes(): array
     {
         if (!isset($this->productAttributes)) {
             $configValue = $this->scopeConfig->getValue(self::XML_PATH_PRODUCT_ATTRIBUTES);
             $productAttributes = $this->json->unserialize($configValue ?? '[]') ?? [];
```

Comment:
Now that we've found it and identified why they do this double null coalesce, there's a few places we can swap it out if you want:

```suggestion
$productAttributes = $this->json->unserialize($configValue ?: '[]');
```

Reply:
I looked into this more. Turns out `$json->unserialize` can return `true` or `false` if those are passed in. I'll change this to more explicitly handle non-array values